### DISCUSSION TITLE
posh-vcpkg is not removed from PowerShell Profile

### DISCUSSION THREAD

**Discussion by NikoMix** [microsoft/vcpkg#18564] [Ideas]
**Describe the bug**
Removing the vcpkg integration does not remove the posh-vcpkg reference of the PowerShell Profile, throwing an error if - as highlighted in the documentation for uninstallation - the folder of vcpkg should be removed.

**Environment**
- OS: Windows
- Compiler: <unknown>

**To Reproduce**
1. .\vcpkg\vcpkg integrate install
2. .\vcpkg\vcpkg integrate remove
3. Delete vcpkg Folder
4. Start PowerShell Console

**Expected behavior**
The Integration is removed from the system and no left-overs from the configuration lead to errors while starting the PowerShell console when the folder is removed

**Failure logs** 
- [PowerShell Issue.txt](https://github.com/microsoft/vcpkg/files/5655154/PowerShell.Issue.txt)

**Comment by JackBoosY** [comment]
cc @BillyONeal

**Comment by BillyONeal** [comment]
`vcpkg integrate remove` only undoes `vcpkg integrate install` -- it has no effect on other integrations. It looks like `integrate powershell` adds a line to `$PROFILE`, an implicit powershell variable. You can remove the powershell integration by deleting the `Import-Module` statement that does that from that script.

I think, but am not certain, that a removal command is intentionally not provided for powershell because that `$PROFILE` script can contain arbitrary powershell code and removing it cleanly wouldn't always be possible, whereas for `integrate install` we are deleting a file we create for that hook outright so it can always be done safely.

**Comment by NikoMix** [comment]
Thank you - I've forgot that the integration had been added through the integrate powershell command. However, would you maybe accept a pull request for cleaning the file? As the posh-vcpkg would not be used/usable by other applications anymore and it's very specific I'd propose that the config is cleared of the Install-Module of the posh-vcpkg and if then only empty files are left in the file, it will get deleted.
I could offer a PR for this change, when I get some time for it.

**Comment by BillyONeal** [comment]
There is the potential concern that that file is arbitrary code etc. so it won't be perfect and this isn't in source control so if we do it wrong things are bad. I think we'll need some kind of validation that there's nothing "interesting" going on in the file but given that we would like such a PR I'd assume.

**Comment by BillyONeal** [comment]
One more thing: We would love to have an if exists check when `vcpkg integrate powershell` installs the integration, which would at least have prevented the error you're indicating above even if a true removal command is difficult.

**Comment by BillyONeal** [comment]
> validation that there's nothing "interesting" going on in the file

We could also generate comment blocks or similar around the inserted region so that we could cleanly remove it too
